.idea/codeStyles/Project.xml

```diff
@@ -0,0 +1,44 @@
+<component name="ProjectCodeStyleConfiguration">
```

Comment:
Is this file truly needed?

Reply:
Well, that's both the files that my beloved sea lion created after opening the project with the latest version.

Looking at https://confluence.jetbrains.com/display/IDEADEV/New+project+code+style+settings+format+in+2017.3 it seems that both files are safe to put in a VCS. While it also states 'the latter [Project.xml] is optional', it then refers it to be created when something differs from the default settings. The 'empty' file being created in my case is probably due to me having some non-standard settings for my default settings, whereas here I rolled back to the default.

Long story short, it seems not to harm anyone and force-pull settings back to default when somebody uses custom settings as their IDE default, thus IMHO it can stay there.

Reply:
Oh interesting, thanks for sharing the update.